modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/YauaaEnrichment.scala

```diff
@@ -91,23 +91,91 @@ final case class YauaaEnrichment(cacheSize: Option[Int]) extends Enrichment {
    * @param userAgent User agent of the event.
    * @return Attributes retrieved thanks to the user agent (if any), as self-describing JSON.
    */
-  def getYauaaContext(userAgent: String): SelfDescribingData[Json] =
-    SelfDescribingData(YauaaEnrichment.outputSchema, parseUserAgent(userAgent).asJson)
+  def getYauaaContext(userAgent: String, headers: List[String]): SelfDescribingData[Json] =
+    SelfDescribingData(YauaaEnrichment.outputSchema, parseUserAgent(userAgent, headers).asJson)
 
   /**
    * Gets the map of attributes retrieved by YAUAA from the user agent.
    * @return Map with all the fields extracted by YAUAA by parsing the user agent.
    *         If the input is null or empty, a map with just the DeviceClass set to Unknown is returned.
    */
-  def parseUserAgent(userAgent: String): Map[String, String] =
+  def parseUserAgent(userAgent: String, headers: List[String]): Map[String, String] =
```

Comment:
The `userAgent` field can come from two places:

- Extracted from the header
- Explicitly set as a field in the event.

By the time we reach this enrichment, we have dropped information about where `userAgent` originally came from.  This cannot be easily changed without quite a big re-write of the enrichment logic.

If the user agent is null or empty, then this request did not original from a browser.  So it is fine to ignore client hint headers, because they are only relevant for web.

We cannot change the function to take a single `headers` field.  Imagine the case of server-side tracking where the headers contains the user agent of the server, but the tp2 payload explicitly sets the user agent of the end client.  In this case, we need to override the `User-Agent` header in the list.  I put in a test for this case, so we don't get any accidental regressions if someone tries to change the code.

Renaming the function is a good idea.  How about `analyzeUserAgent` -- afterall that's what the 3rd `a` in yauaa stands for.